spec/components/document_list_spec.rb

```diff
@@ -533,4 +533,128 @@ def component_name
     assert_select "#{link}[href=\"https://www.actionfraud.police.uk/contact-us\"][rel=\"external\"]", text: "Report Fraud"
     assert_select "#{link}[href=\"/contact-child-benefit-office\"]:not([rel])", text: "Child Benefit"
   end
+
+  it "can accept data attributes to be added to list elements" do
+    render_component(
+      li_data_attributes: {
+        "hello": "world",
+        "another": "example",
+      },
+      items: [
+        {
+          link: {
+            text: "Report Fraud",
+            path: "https://www.actionfraud.police.uk/contact-us",
+            rel: "external",
+          },
+        },
+      ],
+    )
+
+    assert_select "li[data-hello=world]"
+    assert_select "li[data-another=example]"
+  end
+
+  it "can accept data attributes to be added to list elements when GA4 tracking is disabled" do
```

Comment:
This is a good test, but when would this situation occur? I worry that it's a bit misleading - surely if you can pass any data attributes to the list item even when `disable_ga4` is true, it would still be possible to manually pass a GA4 tracking module and still track despite this?

Reply:
@andysellick Good point. I guess I saw it more as a test that `li_data_attributes` code will still works separately from the `ga4_tracking` code, since they are separate features of the component. Though happy to remove if you still think it isn't needed.

Reply:
I think I have clearer understanding of all this now, thanks. In light of that I think the test is necessary, but unless there's a situation where we would disable GA4 but still want to add attributes to this bit of the component (I can't think of one, but happy to consider options) I'd rewrite the code to make the `disable_ga4` option prevent any attributes from being added, regardless of what's passed. Does that seem sensible?

Reply:
Thanks - this is no longer needed so I've removed the test.

Reply:
Good stuff 👍